modules/ROOT/pages/external-oauth-2.0-token-validation-policy.adoc

```diff
@@ -1,43 +1,173 @@
+. The policy extracts the token from the request and sends it to the validation endpoint to verify the integrity of the token.
+. The token validation endpoint returns token metadata, including the client ID of the client application.
+. Using a local database updated with contracts that were previously obtained from Anypoint Platform, the policy verifies whether the client ID has access to the API. 
+. If all the validations are successfully completed, the request is allowed to reach the backend.
 
-You are an Anypoint Platform organization administrator or have permission to create or manage APIs in an environment.
+[[leveraging-TVE-information]]
+=== Leveraging Token Validation Endpoint Information for Mule Applications
+
+When a token validation endpoint successfully validates a token, certain information, such as configurable fields from Mule OAuth provider, are returned to the policy.
+
+The following example illustrates the token validation response from the authorization server:
+
+[source,text,linenums]
+---
+{
+    "uid":"john.doe",
+    "mail":"john.doe@example.com"
+    "token_type":"Bearer",
+    "expires_in":3600,
+    "alias" : ["Jhon", "Jhonny", "Mr Doe"],
+    "address" : {
+        "city": "london", 
+        "road": “abbey road"
+    }
+    
+}
+---
+The fields returned by the OAuth provider are processed by the policy, propagated throughout the Mule flow, and finally exposed to the backend if the application requesting the access uses an HTTP requester.
+
+=== Token Caching
+
+After the system validates a token, that token is cached (by default) until it expires, to improve the performance of the policy. In certain cases, such as in the case of revoked tokens, you might want to minimize or even disable caching.
+
+To control the time taken to cache the tokens before retrying a validation against the token validation endpoint, specify the following property when starting the Mule runtime engine:
+
+`anypoint.platform.max_federation_expiration_time=<expiration time in seconds; specify 0 to disable caching>`
+
+Alternatively, you can specify this parameter in the `wrapper.conf` file.
+ 
+To control the number of tokens that can be cached simultaneously, specify the following property when starting the Mule runtime engine:
+ 
+`anypoint.platform.oauth2_cache_max_size=<amount of tokens to be cached; default value is 10.000>`
+
+Alternatively, you can specify this parameter in the `wrapper.conf` file.
+
+== Configuring the Expose Headers Option
+
+If you configured the `Expose Headers` option in the policy and if the application (for example, out-of-the-box proxies) uses the HTTP requester, the properties are redirected to the backend as headers. 
+
+For each field, the original request to the protected resource is enriched with the HTTP headers using the format: X-AGW- + <key>=<value>. For the example response from the previous section, the following headers are added:
```

Comment:
No need for a colon here:
```suggestion
For each field, the original request to the protected resource is enriched with the HTTP headers using the format X-AGW- + <key>=<value>. For the example response from the previous section, the following headers are added:
```
GLOBALLY avoid referring to any previous or next text; try to make subtopics or sections independent, to avoid issues when one text changes but the other does not.

Reply:
Done! Thanks!